posts/inside-rust/2022-07-29-compiler-team-2022-midyear-report.md

```diff
@@ -0,0 +1,797 @@
+
+</table>
+
+Here are some trends I noted from looking at the table:
+
+The overwhelming majority of our ambitions, 19 out of 22, reported *some*
+amount of progress. That's great, esepcially given that several of those
+ambitions were mere aspirations that explicitly had no resources attached to
+them at the start of the year.
+
+Less than a quarter of the ambitions, 6 out of 22, said that their overall
+problem would be resolved, or the bulk of the problem addressed, within the next
+six months. Ten others said they predicted the important components of their
+solution would be available within the next two years, so that's a total of 16
+out of 22 that expect some kind of solution within two years.
+
+  * Note: One respondant noted that the two year threshold embedded in the
+    survey was an "odd timeframe." I do not disagree with that remark. The
+    reality is that I was trying to find a happy medium between two extremes:
+    short timeframes don't allow enough time for significant work to be
+    accomplished, while excessively long timeframes (e.g. greater than five
+    years) are very difficult to predict realistically. But the two year
+    time frame was drawn somewhat out of a hat.
+
+Of the nineteen ambitions that reported some progress so far this year, fourteen
+of them (14/19) also said that at the start of the year, they had either *no
+goals* or *no plan* to achieve those goals. Ten of those fourteen say they now
+have a better understanding of the problem, and five of those fourteen say they
+now have a plan.
+
+  * Here is a question I want to put forward to our communtiy: Do we, as a
+    project, need to put more energy into upfront planning? Or, given our
+    volunteer nature, will that just distract people from "the fun stuff" and
+    potentially burn out contributors? In other words: is this 5/14 "ambitions
+    that spent time making a concrete plan", is that a value we
+    should work on growing? Or should we just monitor it, and not actively try to
+    change it?
+
+  * Note: It is possible that my survey was itself somewhat flawed, so we need
+    to be careful in how we interpret this data. My intention was to try to
+    capture how well structured a given ambition's plan was by asking if it had
+    established milestones and/or a schedule. However, the survey results
+    include cases where the respondant did *not* indicate that they had made a
+    plan, but they *did* say that this year they did establish milestones.
+    Should I have interpreted that as implicitly saying that there *was* a plan
+    established? Maybe we need to restructure future surveys to make these
```

Comment:
It could be interpreted as retroactively saying that something was a big enough improvement to call it hitting a milestone, without having a concrete plan before those milestones were achieved. And that there is no plan on what milestones to hit next.

Reply:
exactly, that is an excellent example of the kind of way that one might have had very good reason to click the checkbox saying that milestones were hit while still leaving the other checkboxes unset.

I'll incorporate your example into the text

Reply:
Added in https://github.com/rust-lang/blog.rust-lang.org/pull/1017/commits/0f0bf0239804a895fde8da65dbf9d5315d3d8e75